frontend/src/main/java/com/google/tripmeout/frontend/servlets/ServletUtil.java

```diff
@@ -0,0 +1,45 @@
+package com.google.tripmeout.frontend.servlets;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.Streams;
+import com.google.common.flogger.FluentLogger;
+import com.google.gson.Gson;
+import com.google.gson.JsonParseException;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Static utilities that provide functionality that may be shared between servlets.
+ */
+public final class ServletUtil {
+  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+
+  public static <T> Optional<T> extractFromRequestBody(
+      HttpServletRequest request, Gson gson, Class<T> clazz) throws IOException {
```

Comment:
As is evidenced by your test's lack of type checking on some generics, you might want to have this method overloaded version of this that accepts `java.lang.reflect.Type` instead of `Class<T>` to allow type token usage for generics.